clients/drcachesim/tracer/raw2trace.h

```diff
@@ -1122,7 +1228,8 @@ class raw2trace_t {
     // instruction.
     std::string
     write(raw2trace_thread_data_t *tdata, const trace_entry_t *start,
-          const trace_entry_t *end, std::vector<app_pc> decode_pcs = {});
+          const trace_entry_t *end, app_pc *decode_pcs = nullptr,
+          size_t decode_pcs_size = 0);
```

Comment:
These other changes are also not in the PR description: please add.

What % of speedup is from these?

Reply:
This was resolved but I don't see this in the updated PR description?  I'm seeing 4 different changes here:

1. lz4
2. bitset instead of set for encoding history
3. not passing vector by value
4. keeping bool for whether delayed branch container is empty

I think these should be explicitly listed in the PR description.  Otherwise it is confusing reading the code and puzzling over how these relate to lz4 or the bitset.  Maybe at least some of these should be in a separate PR; maybe the lz4 in one PR, and the other 3 in another.

Reply:
ok. 2-4 will be in separate PR